### DISCUSSION TITLE
window_control not working

### DISCUSSION THREAD

**Discussion by llamafilm** [llamafilm/tesla-http-proxy-addon#21] [General]
The `window_control` vent command is not working for me.  It returns HTTP 200 response with no content.  I'm curious if it works for anyone else using this proxy?  All other commands are working fine here.  I have a 2021 M3 in North America.
I opened an issue: https://github.com/teslamotors/vehicle-command/issues/210

**Comment by baylanger** [comment]
Have you noticed [this comment](https://github.com/teslamotors/vehicle-command/issues/188#issuecomment-1951454182) and the one below?

**Reply by baylanger** [reply]
Hmmm... I had not first Search in T code for Vent and the command seems already in there, otherwise it probably wouldn't get to the point to send the command.

**Reply by baylanger** [reply]
You are way better to answer the following... Maybe [this](https://github.com/teslamotors/vehicle-command/blob/b57e560979c9d43f0ed544be2ae6dd4438f86d95/pkg/protocol/protobuf/car_server.proto#L328) can help answer the question if lat/long is required or not for vent?

Searching for window and vent in the T code doesn't reveal much more.

**Reply by baylanger** [reply]
Eureka. Just stumble on [this issue](https://github.com/teslamotors/vehicle-command/issues/43) for the missing window_control.

**Reply by llamafilm** [reply]
Thanks, I actually saw that but skimmed over because I saw SDK in the title.  But your'e right, this is the same issue.  So I take it this isn't working for you either?

**Reply by baylanger** [reply]
The cobbler is badly shod... I greatly appreciate your work. Perhaps I should spend less time helping people here but I'm doing that hoping you can focus in the coding. Bottom line, I never completed the manual install.

While writing ^^ , I noticed the PR was merged and tesla-v3.20 was released. I have no more excuse not upgrade and complete the setup. I'll post back when I'm done!

**Reply by llamafilm** [reply]
Haha, I do appreciate your running interference on the support issues!

**Reply by baylanger** [reply]
I look in the Tesla add-on config for my car to try to enable the proxy... I ended up deleting my car and adding it again to notice the option for the proxy.

Vent doesn't work for my car either:

  File "/config/custom_components/tesla_custom/cover.py", line 151, in async_open_cover
    await self._car.vent_windows()
  File "/usr/local/lib/python3.12/site-packages/teslajsonpy/car.py", line 1184, in vent_windows
    data = await self._send_command("WINDOW_CONTROL", command="vent", lat=0, long=0)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

lol "interference" is definitely appropriate, a few times I should have taken more time to read and Discussions should be use when not an issue :)
